src/Bonsai.ML.PointProcessDecoder/PointProcessModelDisposable.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Threading;
+using PointProcessDecoder.Core;
+
+namespace Bonsai.ML.PointProcessDecoder;
+
+internal sealed class PointProcessModelDisposable(PointProcessModel model, IDisposable disposable) : IDisposable
+{
+    private IDisposable? resource = disposable ?? throw new ArgumentNullException(nameof(disposable));
+    /// <summary>
+    /// Gets a value indicating whether the object has been disposed.
+    /// </summary>
+    public bool IsDisposed => resource == null;
```

Comment:
```suggestion
    public bool IsDisposed => resource is null;
```

Not a big deal as I have not changed everything myself, but nowadays `is` tends to be preferred to `==` as it forces reference nullability and protects against operator overloading of the `==` operator.